### PR TITLE
DOC: Update docs of jnp.all and jnp.any

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -434,9 +479,54 @@ def _reduce_any(a: ArrayLike, axis: Axis = None, out: None = None,
...
+      with size 1.
+    where: int or array of boolean dtype, default=None. The elements to be used
+      in the test. Array should be broadcast compatible to the input.
+    out: Unused by JAX.
+
+  Returns:
+    An array of boolean values.
+
+  Examples:
+
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:503]: Remove blank line

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -422,9 +422,54 @@ def _reduce_all(a: ArrayLike, axis: Axis = None, out: None = None,
...
 def all(a: ArrayLike, axis: Axis = None, out: None = None,
         keepdims: bool = False, *, where: ArrayLike | None = None) -> Array:
+  r"""Test whether all array elements along a given axis evaluate to True.
+
+  JAX implementation of :func:`numpy.all`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the test to be performed
+      to evaluate for True values. If None, tests along all the axes.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:435]: Fix grammar in the sentence starting with "Axis"

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good! Just a couple small comments

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -422,9 +422,54 @@ def _reduce_all(a: ArrayLike, axis: Axis = None, out: None = None,
...
+      with size 1.
+    where: int or array of boolean dtype, default=None. The elements to be used
+      in the test. Array should be broadcast compatible to the input.
+    out: Unused by JAX.
+
+  Returns:
+    An array of boolean values.
+
+  Examples:
+
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Removed

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Removed

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Fixed and squashed into single commit.

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

**Comment by jakevdp** [comment]
Can you please squash the commits?

**Comment by rajasekharporeddy** [comment]
Squashed the commits.

**Review by jakevdp — APPROVED** [review]
